### PR TITLE
Adds Document Logic 

### PR DESCRIPTION
**Pull request by Moejoe90**

### CODE REVIEW

File: app/lib/api/eto/EtoFileApi.interfaces.ts [hunk 1]
```diff
@@ -15,41 +16,50 @@ export interface IEtoGeneratedFile {
...
-  };
+  etoTemplates: TEtoDocumentTemplates;
+  uploadedDocuments: TEtoDocumentTemplates;
+  stateInfo?: { [key in ISimpleFileInfo]: etoDocumentType[] } &
+    { [key in IComplextFileInfo]: { [key in EtoState]: etoDocumentType[] } };
 }
+
+export type TEtoDocumentTemplates = { [key: string]: IEtoDocument };
+
+export const immutableDocumentName: { [key: string]: string } = {
```
Comment (trival) [inline app/lib/api/eto/EtoFileApi.interfaces.ts]: Shouldn't these strings be translated?

File: app/modules/eto-documents/sagas.ts [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    yield put(actions.etoDocuments.hideIpfsModal());
+
+    yield neuCall(
+      ensurePermissionsArePresent,
+      [UPLOAD_IMMUTABLE_DOCUMENT],
+      formatIntlMessage("eto.modal.submit-description"),
+    );
+
+    const matchingDocument = yield neuCall(getDocumentOfTypePromise, documentType);
+    apiEtoFileService.deleteSpecificEtoDocument(matchingDocument.ipfsHash);
```
Comment (trival) [inline app/modules/eto-documents/sagas.ts]: does that not need to be yielded as well?

File: app/modules/eto-documents/sagas.ts [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    yield put(actions.etoDocuments.hideIpfsModal());
+
+    yield neuCall(
+      ensurePermissionsArePresent,
+      [UPLOAD_IMMUTABLE_DOCUMENT],
+      formatIntlMessage("eto.modal.submit-description"),
+    );
+
+    const matchingDocument = yield neuCall(getDocumentOfTypePromise, documentType);
+    if (matchingDocument) apiEtoFileService.deleteSpecificEtoDocument(matchingDocument.ipfsHash);
```
Comment (trival) [inline app/modules/eto-documents/sagas.ts]: don't you need to yield this service call as well?

Comment (Moejoe90) [reply app/modules/eto-documents/sagas.ts]: Absolutly

File: app/lib/api/eto/EtoFileApi.interfaces.ts (hunk 1 above)
Comment (Moejoe90) [reply app/lib/api/eto/EtoFileApi.interfaces.ts]: I would keep them like this for now, and move them somewhere else later on

File: app/lib/api/client/AuthBinaryHttpClient.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
+/**
+ * Wraps the binary http api with authorization header injection
+ * collected from localstorage
+ */
+import { symbols } from "../../../di/symbols";
+
```
[the remark this answers is not in the document]
Comment (Moejoe90) [reply app/lib/api/client/AuthBinaryHttpClient.ts]: done

File: app/components/shared/singleColDocumentWidget.tsx [hunk 5]
```diff
@@ -1,39 +1,68 @@
 import * as cn from "classnames";
 import * as React from "react";
 import { Col } from "reactstrap";
+import * as styles from "./singleColDocumentWidget.module.scss";
```
[the remark this answers is not in the document]
Comment (Moejoe90) [reply app/components/shared/singleColDocumentWidget.tsx]: done

File: app/components/eto/shared/EtoFileIpfsModal.tsx [hunk 6]
```diff
@@ -3,12 +3,15 @@ import { FormattedMessage } from "react-intl-phraseapp";
...
 
 import { actions } from "../../../modules/actions";
-import { selectFileUploadAction, selectIsIpfsModalOpen } from "../../../modules/eto-flow/selectors";
 import { appConnect } from "../../../store";
 import { ModalComponentBody } from "../../modals/ModalComponentBody";
 import { ButtonArrowRight } from "../../shared/Buttons";
 
 import * as ipfsImage from "../../../assets/img/ipfs.png";
+import {
+  selectFileUploadAction,
```
[the remark this answers is not in the document]
Comment (Moejoe90) [reply app/components/eto/shared/EtoFileIpfsModal.tsx]: done

File: app/components/eto/shared/EtoAddDocument.tsx [hunk 7]
```diff
@@ -2,32 +2,32 @@ import * as React from "react";
 import Dropzone from "react-dropzone";
 import { compose } from "redux";
 
-import { TEtoUploadFile } from "../../../lib/api/eto/EtoFileApi.interfaces";
 import { IKycFileInfo, TKycRequestType } from "../../../lib/api/KycApi.interfaces";
 import { actions } from "../../../modules/actions";
 import { appConnect } from "../../../store";
 
+import { etoDocumentType } from "../../../lib/api/eto/EtoFileApi.interfaces";
```
[the remark this answers is not in the document]
Comment (Moejoe90) [reply app/components/eto/shared/EtoAddDocument.tsx]: done

File: app/components/eto/EtoDashboard.tsx [hunk 8]
```diff
@@ -30,13 +30,15 @@ import {
   selectCombinedEtoCompanyData,
   selectCompanyData,
   selectEtoData,
-  selectEtoLoadingData,
   selectEtoState,
+} from "../../modules/eto-flow/selectors";
+
```
[the remark this answers is not in the document]
Comment (Moejoe90) [reply app/components/eto/EtoDashboard.tsx]: done

File: app/components/Documents.tsx [hunk 9]
```diff
@@ -3,46 +3,83 @@ import { FormattedMessage } from "react-intl";
...
 import { ETOAddDocuments } from "./eto/shared/EtoAddDocument";
 import { EtoFileIpfsModal } from "./eto/shared/EtoFileIpfsModal";
 import { LayoutAuthorized } from "./layouts/LayoutAuthorized";
 import { DocumentTile } from "./shared/Document";
 import { LoadingIndicator } from "./shared/LoadingIndicator";
 import { SectionHeader } from "./shared/SectionHeader";
-import { SingleColDocumentsWidget } from "./shared/singleColDocumentWidget";
+import { SingleColDocuments } from "./shared/singleColDocumentWidget";
 
+import * as cn from "classnames";
```
[the remark this answers is not in the document]
Comment (Moejoe90) [reply app/components/Documents.tsx]: done

**Pull request merged**